Assets/MixedRealityToolkit.Services/InputSystem/FocusProvider.cs

```diff
@@ -1050,11 +1058,20 @@ private void UpdateFocusedObjects()
                 }
             }
 
+            // Early out if there have been no focus changes
+            if (pendingPointerSpecificFocusChange.Count == 0)
+            {
+                return;
+            }
+
             // ... but now we trim out objects whose overall focus was maintained the same by a different pointer:
 
             foreach (var pointer in pointers)
             {
-                pendingOverallFocusExitSet.Remove(pointer.CurrentPointerTarget);
+                if (pointer.CurrentPointerTarget != null)
```

Comment:
Is this null check necessary? It's not performed for the pendingOverallFocusEnterSet.

Reply:
It is. I found out the hard way: Dictionary will throw an exception if you try to remove a null key.